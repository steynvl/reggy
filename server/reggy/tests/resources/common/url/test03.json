{
  "type": "URL",
  "information": {
    "schemes": "feed;http;https;www.",
    "portNumbers": "Require port number",
    "specOptionalPortNumbers": "",
    "specRequiredPortNumbers": "",
    "username": "Allow any user name",
    "specUserNames": "",
    "password": "No password",
    "domainName": "Allow any domain name",
    "specDomainNames": "",
    "specificTld": "",
    "subdomainOnSpecDomain": "",
    "folders": "No folders",
    "minFolderDepth": "",
    "maxFolderDepth": "",
    "specFoldersOnly": "",
    "fileNames": "No file names",
    "specExtensions": "",
    "specFileNames": "",
    "optionalFileNames": false,
    "parameters": "No parameters",
    "specParameters": ""
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
