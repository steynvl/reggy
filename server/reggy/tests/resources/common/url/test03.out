Java
Pattern regex = Pattern.compile("(?:(?:feed|https?)://|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+@)?(?:[\\da-z-]++\\.)+[a-z]{2,63}+:\\d{1,5}+/?");
(?:(?:feed|https?)://|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+@)?(?:[\\da-z-]++\\.)+[a-z]{2,63}+:\\d{1,5}+/?
Perl
my $regex = /(?:(?:feed|https?):\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]++\.)+[a-z]{2,63}+:\d{1,5}+\/?/;
(?:(?:feed|https?):\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]++\.)+[a-z]{2,63}+:\d{1,5}+\/?
POSIX
(?:(?:feed|https?)://|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]++\.)+[a-z]{2,63}+:\d{1,5}+/?
(?:(?:feed|https?)://|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]++\.)+[a-z]{2,63}+:\d{1,5}+/?
Python
regex = re.compile(r'(?:(?:feed|https?)://|www\.)(?:[]\d!"#$%&\'()*+,./:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]+\.)+[a-z]{2,63}:\d{1,5}/?')
(?:(?:feed|https?)://|www\.)(?:[]\d!"#$%&\'()*+,./:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]+\.)+[a-z]{2,63}:\d{1,5}/?
JavaScript
const regex = /(?:(?:feed|https?):\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]+\.)+[a-z]{2,63}:\d{1,5}\/?/;
(?:(?:feed|https?):\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]+\.)+[a-z]{2,63}:\d{1,5}\/?
PHP
$regex = '/(?:(?:feed|https?):\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]++\.)+[a-z]{2,63}+:\d{1,5}+\/?/';
(?:(?:feed|https?):\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]++\.)+[a-z]{2,63}+:\d{1,5}+\/?
Golang
regex, _ := regexp.Compile("(?:(?:feed|https?)://|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+@)?(?:[\\da-z-]+\\.)+[a-z]{2,63}:\\d{1,5}/?")
(?:(?:feed|https?)://|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+@)?(?:[\\da-z-]+\\.)+[a-z]{2,63}:\\d{1,5}/?
Rust
let re = Regex::new(r"(?:(?:feed|https?)://|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]+\.)+[a-z]{2,63}:\d{1,5}/?").unwrap();
(?:(?:feed|https?)://|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]+\.)+[a-z]{2,63}:\d{1,5}/?
C#
var re = new Regex(@"(?:(?:feed|https?)://|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]+\.)+[a-z]{2,63}:\d{1,5}/?");
(?:(?:feed|https?)://|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+@)?(?:[\da-z-]+\.)+[a-z]{2,63}:\d{1,5}/?
Scala
val re = "(?:(?:feed|https?)://|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+@)?(?:[\\da-z-]++\\.)+[a-z]{2,63}+:\\d{1,5}+/?".r
(?:(?:feed|https?)://|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+@)?(?:[\\da-z-]++\\.)+[a-z]{2,63}+:\\d{1,5}+/?
Kotlin
val regex = Regex("(?:(?:feed|https?)://|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+@)?(?:[\\da-z-]++\\.)+[a-z]{2,63}+:\\d{1,5}+/?")
(?:(?:feed|https?)://|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+@)?(?:[\\da-z-]++\\.)+[a-z]{2,63}+:\\d{1,5}+/?